### PR TITLE
react-file-uploads

### PR DESCRIPTION
**Pull request by yuliia-p**
Screen Shot 2022-09-08 at 4 46 56 PM
Screen Shot 2022-09-08 at 4 49 39 PM

Screen Shot 2022-09-08 at 4 47 04 PM

### CODE REVIEW

File: react-file-uploads/client/app.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     * Catch any error in the promise chain.
+     *
+     * References:
+     * https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData
+     * https://developer.mozilla.org/en-US/docs/Web/API/FormData/append
+     * https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_a_file
+     * https://reactjs.org/docs/uncontrolled-components.html#the-file-input-tag
+     * https://reactjs.org/docs/refs-and-the-dom.html
+     */
+  }
```
Comment (thebearingedge) [inline react-file-uploads/client/app.jsx:48]: the client code also needs to be completed

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: react-file-uploads/client/app.jsx (hunk 1 above)
Comment (yuliia-p) [reply react-file-uploads/client/app.jsx]: Screen Shot 2022-09-08 at 5 04 15 PM
Screen Shot 2022-09-08 at 5 04 11 PM
Screen Shot 2022-09-08 at 5 04 07 PM

File: react-file-uploads/client/app.jsx [hunk 2]
```diff
@@ -16,6 +16,22 @@ export default class App extends React.Component {
   }
 
   handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData();
+    formData.append(this.state.caption, this.fileInputRef.current.files[0]);
```
Comment (thebearingedge) [inline react-file-uploads/client/app.jsx:21]: The formData object needs two fields appended to it, one named "caption", and one named "image". Currently you are appending only one field where the name _is_ the caption value and the value is the image file.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: react-file-uploads/client/app.jsx (hunk 2 above)
Comment (yuliia-p) [reply react-file-uploads/client/app.jsx:21]: Screen Shot 2022-09-08 at 5 22 33 PM

Screen Shot 2022-09-08 at 5 22 51 PM

Screen Shot 2022-09-08 at 5 26 27 PM

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**